### DISCUSSION TITLE
GoControl CECOMINOD016164 HUSBZB-1 USB Hub - any hope?

### DISCUSSION THREAD

**Discussion by doron1** [Koenkk/zigbee2mqtt#13324] [Devices]
Hi,
I have [this adapter](https://smile.amazon.com/gp/product/B01GJ826F8/) and having issues trying to make it work under zigbee2mqtt (running as addon on HA OS). The adapter _sort of_ works with ZHA but it's kind of hit and miss and very unstable; with zigbee2mqtt I just can't get the addon to fully initialize. It sits after "Starting zigbee-herdsman (0.14.40)" and just hangs there.
I tried everything I could find on this, including [this](https://github.com/walthowd/husbzb-firmware) and many others; still can't make it work.

I can continue with my tinkering, but since the adapter is not on either the list of supported or unsupported adapters, I thought I'd ask here - does anyone have this working with zigbee2mqtt? Is there hope? Thanks for any insight.

**Comment by DubhAd** [comment]
It should be [one of the experimental EZSP](https://www.zigbee2mqtt.io/guide/adapters/#experimental) options, and it's mentioned [in this issue](https://github.com/Koenkk/zigbee-herdsman/issues/319#issuecomment-789757269).

As that issue says, it's not recommended. I'd buy something CC2652 based.

**Comment by kirovilya** [comment]
@doron1 @DubhAd  What is the firmware version of your coordinator?
latest versions of z2m should work quite well with EZSP

**Reply by doron1** [reply]
> @doron1 @DubhAd What is the firmware version of your coordinator?

I'm not sure. I couldn't read it out even with "ncp.py" from [https://github.com/walthowd/husbzb-firmware](https://github.com/walthowd/husbzb-firmware), which barfed with an error ([this one](https://github.com/walthowd/husbzb-firmware/issues/31)) if run natively, and printed an invalid version string after I modded it to bypass the Python exception :-(
Did not yet have time to actually hack the protocol further.

I did try the flash function of the same script, which surprisingly declared success (installing f/w 6.7.8). Still, same results with z2m.

**Reply by kirovilya** [reply]
Version 6.7.8 should start working successfully. You may need to specify the port speed as indicated here https://github.com/walthowd/husbzb-firmware/issues/31#issuecomment-912633373

**Comment by Hedda** [comment]
FYI, since EZSP support is still experimental recommend trying to keep discussions to that issue-> https://github.com/Koenkk/zigbee-herdsman/issues/319

PS: Many more other tips on that specific adapter here -> https://community.home-assistant.io/t/itead-zigbee-3-0-usb-dongle-stick-adapter-based-on-silicon-labs-efr32mg21/271144

PPS: ITead have just released a new EFR32MG21 based USB adapter to replace that and solving its HW issues, see -> https://github.com/Koenkk/zigbee2mqtt/discussions/13373
